pkgs/stdenv/generic/make-derivation.nix

```diff
@@ -41,7 +41,7 @@ let
               else x;
         in
           makeDerivationExtensible
-            (self: let super = rattrs self; in super // f self super);
+            (self: let super = rattrs self; in super // (if builtins.isFunction f0 then f self super else f0));
```

Comment:
Probably not worth the cycles.
Functors are fundamentally ambiguous here anyway, so let's just do whatever evaluates ever so slightly faster.

Technically omitting this is a breaking change, but getting functors in here is highly unlikely, as the functions that are actually passed are lambdas in basically every single case I've ever seen.
Grepping for `overrideAttrs [a-z]` in nixpkgs only yields definitions that forward another overriding method, or an occasional `let` binding for syntactic convenience; never something from `makeOverridable` or `callPackage` or whatever.

Just note the breaking change in the commit message, in case someone does have to bisect to find the problem, which I find highly unlikely.

Reply:
is this ok in the msg `breaks overriding with __functor`